### PR TITLE
Add convenient methods to rule/issue testing utilities

### PR DESCRIPTION
**Pull request by simonbrandhof**

### CODE REVIEW

File: server/sonar-db-dao/src/main/java/org/sonar/db/issue/IssueTesting.java [hunk 1]
```diff
@@ -19,32 +19,59 @@
...
-   * Full IssueDto used to feed database with fake data. Tests must not rely on the
-   * field contents declared here. They should override the fields they need to test,
-   * for example:
-   * <pre>
-   *   issueDao.insert(dbSession, IssueTesting.newDto(rule, file, project).setStatus(Issue.STATUS_RESOLVED).setResolution(Issue.RESOLUTION_FALSE_POSITIVE));
-   * </pre>
+   * @deprecated use newIssue(...)
    */
+  @Deprecated
   public static IssueDto newDto(RuleDto rule, ComponentDto file, ComponentDto project) {
```
Comment (SonarTech) [inline server/sonar-db-dao/src/main/java/org/sonar/db/issue/IssueTesting.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'

File: server/sonar-db-dao/src/test/java/org/sonar/db/rule/RuleTesting.java [hunk 2]
```diff
@@ -46,60 +50,128 @@ private RuleTesting() {
...
+  public static RuleMetadataDto newRuleMetadata() {
+    return randomizeFields(new RuleMetadataDto()
+      .setRuleId(nextInt(1_000_000))
+      .setOrganizationUuid("org_" + randomAlphanumeric(5)));
+  }
+
+  public static RuleMetadataDto newRuleMetadata(RuleDefinitionDto rule, OrganizationDto organization) {
+    return randomizeFields(new RuleMetadataDto()
+      .setRuleId(rule.getId())
+      .setOrganizationUuid(organization.getUuid()));
```
Comment (bartfastiel) [inline server/sonar-db-dao/src/test/java/org/sonar/db/rule/RuleTesting.java]: I would suggest to always randomize all fields and afterwards overwrite the required fields.

File: server/sonar-db-dao/src/test/java/org/sonar/db/rule/RuleDbTester.java [hunk 3]
```diff
@@ -36,11 +37,58 @@ public RuleDbTester(DbTester db) {
     this.db = db;
   }
 
-  public RuleDefinitionDto insertRule(RuleDefinitionDto ruleDto) {
```
Comment (bartfastiel) [inline server/sonar-db-dao/src/test/java/org/sonar/db/rule/RuleDbTester.java]: 👍

File: server/sonar-db-dao/src/test/java/org/sonar/db/rule/RuleTesting.java [hunk 4]
```diff
@@ -46,60 +50,128 @@ private RuleTesting() {
...
+   * @deprecated use newRule(...)
    */
+  @Deprecated
   public static RuleDto newXooX1() {
     return newDto(XOO_X1).setLanguage("xoo");
   }
 
   /**
-   * Create a RuleDto representing the definition of the rule Xoo1 of language Xoo with some user defined fields.
+   * @deprecated use newRule(...)
```
Comment (bartfastiel) [inline server/sonar-db-dao/src/test/java/org/sonar/db/rule/RuleTesting.java]: 👍

**Pull request closed without merging**